int32 id
string label
int32[] box
string name